### PR TITLE
Fix Short Circuiting Update

### PR DESCRIPTION
**Pull request by jamiebrynes7**
#### Description
Look at first commit only. The latter is just generated code.
#### Tests
Compiled both projects - ran playground, updates were still serialised. Could see color changing on the spinners on the worker so change fixed what was intended.
#### Documentation
N/A
#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: test-project/Assets/Generated/Source/improbable/gdk/tests/ExhaustiveBlittableSingular.cs [hunk 1]
```diff
@@ -500,13 +486,136 @@ public static Generated.Improbable.Gdk.Tests.ExhaustiveBlittableSingular.Update
...
                         var value = global::Generated.Improbable.Gdk.Tests.SomeType.Serialization.Deserialize(obj.GetObject(17));
-                        component.Field17 = value;
                         update.Field17 = new global::Improbable.Gdk.Core.Option<global::Generated.Improbable.Gdk.Tests.SomeType>(value);
                     }
                     
                 }
                 return update;
             }
+
+            public static void ApplyUpdate(global::Improbable.Worker.Core.SchemaObject obj, ref Generated.Improbable.Gdk.Tests.ExhaustiveBlittableSingular.Component component)
```
Comment (samiwh) [inline test-project/Assets/Generated/Source/improbable/gdk/tests/ExhaustiveBlittableSingular.cs]: obj should probably be update

File: workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentConversionGenerator.tt [hunk 2]
```diff
@@ -162,32 +162,32 @@ namespace <#= qualifiedNamespace #>
...
-                            handle = ReferenceTypeProviders.UpdatesProvider.Allocate(World)
-                        };
-                        ReferenceTypeProviders.UpdatesProvider.Set(updatesComponent.handle, new List<<#= componentNamespace #>.Update>());
-                        updates = updatesComponent.Updates;
-                        entityManager.AddComponentData(entity, updatesComponent);
-                    }
-
-                    updates.Add(update);
-
-                    data.DirtyBit = false;
```
Comment (firtina-improbable) [inline workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentConversionGenerator.tt]: should the dirtybit still get affected here?

Comment (firtina-improbable) [reply workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentConversionGenerator.tt]: I guess there is no `data` left so sounds like not

Comment (jamiebrynes7) [reply workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentConversionGenerator.tt]: You are correct!

**Pull request merged**